src/main/java/pl/kamann/services/ConfirmUser.java

```diff
@@ -0,0 +1,61 @@
+package pl.kamann.services;
+
+import jakarta.mail.MessagingException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import pl.kamann.config.codes.AuthCodes;
+import pl.kamann.config.exception.handler.ApiException;
+import pl.kamann.entities.appuser.AppUser;
+import pl.kamann.repositories.AppUserRepository;
+import pl.kamann.services.email.ConfirmationEmail;
+
+import java.util.UUID;
+
+@Component
+@RequiredArgsConstructor
+public class ConfirmUser {
+    private final AppUserRepository appUserRepository;
+    private final ConfirmationEmail emailSender;
+    @Value("${confirmation.link}")
+    private String confirmationLink;
+
+    public String generateConfirmationLink(String token, String confirmationLink) {
+        return confirmationLink + token;
+    }
+
+    public String generateConfirmationToken() {
+        return UUID.randomUUID().toString();
+    }
+
+    public void sendConfirmationEmail(AppUser appUser) {
+        try {
+            emailSender.sendConfirmationEmail(appUser.getEmail(), generateConfirmationLink(appUser.getConfirmationToken(), confirmationLink));
+        } catch (MessagingException e) {
+            throw new ApiException(
+                    "Error sending the confirmation email.",
+                    HttpStatus.INTERNAL_SERVER_ERROR,
+                    HttpStatus.INTERNAL_SERVER_ERROR.name()
+            );
+        }
+    }
+
+    public void confirmUserAccount(String token) {
+        appUserRepository.findByConfirmationToken(token)
```

Comment:
The ConfirmationToken table will be required to properly remove token after confirming the user (will be done with AppUser refactor).